python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -517,3 +509,102 @@ def to_arrow(self):
                 arrow_polygons,
             ],
         )
+
+    def _align_to_index(
+        self: T,
+        index: ColumnLike,
+        how: str = "outer",
+        sort: bool = True,
+        allow_non_unique: bool = False,
+    ) -> T:
+        """
+        The values in the newly aligned columns will not change,
+        only positions in the union offsets and type codes.
+        """
+        aligned_union_offsets = (
+            self._column._meta.union_offsets._align_to_index(
+                index, how, sort, allow_non_unique
+            )
+        ).astype("int32")
+        aligned_union_offsets[
+            aligned_union_offsets.isna()
+        ] = Feature_Enum.NONE.value
+        aligned_input_types = self._column._meta.input_types._align_to_index(
+            index, how, sort, allow_non_unique
+        ).astype("int8")
+        aligned_input_types[
+            aligned_input_types.isna()
+        ] = Feature_Enum.NONE.value
+        column = GeoColumn(
+            (
+                self._column.points,
+                self._column.mpoints,
+                self._column.lines,
+                self._column.polygons,
+            ),
+            {
+                "input_types": aligned_input_types,
+                "union_offsets": aligned_union_offsets,
+            },
+        )
+        return GeoSeries(column)
+
+    def align(self, other):
+        """
+        Align the rows of two GeoSeries using outer join.
+
+        Parameters
+        ----------
+        other: GeoSeries
+
+        Returns
+        -------
+        (left, right) : GeoSeries
+            Pair of aligned GeoSeries
+
+        Examples
+        --------
+        >>> polygons = gpd.GeoSeries([
+            Polygon(((-8, -8), (-8, 8), (8, 8), (8, -8))),
+            Polygon(((-2, -2), (-2, 2), (2, 2), (2, -2))),
+        ])
+        >>> polygon = gpd.GeoSeries(polygons[0])
+        >>> polygon.align(polygons)
+        (0    POINT (-8.00000 -8.00000)
+        1                         None
+        dtype: geometry, 0    POINT (-8.00000 -8.00000)
+        1    POINT (-2.00000 -2.00000)
+        dtype: geometry)
+
+        >>> polygons_right = gpd.GeoSeries([
+            Point((-2, -2)),
+            Point((-8, -8)),
```

Comment:
Why is this called polygons_right if it contains `Point`s, not `Polygon`s?

Reply:
Copy/pasto